connect/src/com/telenor/mobileconnect/MobileConnectSdkProfile.java

```diff
@@ -0,0 +1,316 @@
+package com.telenor.mobileconnect;
+
+import android.content.Context;
+import android.net.Uri;
+import android.telephony.TelephonyManager;
+import android.text.TextUtils;
+import android.util.Base64;
+
+import com.squareup.okhttp.HttpUrl;
+import com.telenor.connect.AbstractSdkProfile;
+import com.telenor.connect.id.ConnectAPI;
+import com.telenor.connect.id.ConnectIdService;
+import com.telenor.connect.id.ConnectTokensTO;
+import com.telenor.connect.id.TokenStore;
+import com.telenor.connect.id.UserInfo;
+import com.telenor.connect.utils.ConnectUrlHelper;
+import com.telenor.connect.utils.RestHelper;
+import com.telenor.mobileconnect.id.MobileConnectAPI;
+import com.telenor.mobileconnect.operatordiscovery.OperatorDiscoveryAPI;
+import com.telenor.mobileconnect.operatordiscovery.OperatorDiscoveryConfig;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import retrofit.Callback;
+import retrofit.ResponseCallback;
+import retrofit.RetrofitError;
+import retrofit.client.Response;
+
+public class MobileConnectSdkProfile extends AbstractSdkProfile {
+
+    private OperatorDiscoveryConfig operatorDiscoveryConfig;
+    private volatile OperatorDiscoveryAPI.OperatorDiscoveryResult operatorDiscoveryResult;
+    private volatile OperatorDiscoveryAPI operatorDiscoveryApi;
+
+    public MobileConnectSdkProfile(
+            Context context,
+            final OperatorDiscoveryConfig operatorDiscoveryConfig,
+            boolean useStaging,
+            boolean confidentialClient) {
+        super(context, useStaging, confidentialClient);
+        this.operatorDiscoveryConfig = operatorDiscoveryConfig;
+    }
+
+    @Override
+    public HttpUrl getApiUrl() {
+        String host = operatorDiscoveryResult.getMobileConnectApiUrl().host();
+        if (useStaging) {
+            // will have no effect on non-Telenor subscribers
+            host = host.replace(
+                    "connect.telenordigital.com",
+                    "connect.staging.telenordigital.com");
+        }
+        HttpUrl.Builder builder = new HttpUrl.Builder();
+        builder
+                .scheme(operatorDiscoveryResult.getMobileConnectApiUrl().scheme())
+                .host(host);
+        for (String seg : operatorDiscoveryResult.getMobileConnectApiUrl().pathSegments()) {
+            builder.addPathSegment(seg);
+        }
+        return builder.build();
+    }
+
+    @Override
+    public String getClientId() {
+        return operatorDiscoveryResult.getClientId();
+    }
+
+    @Override
+    public String getClientSecret() {
+        return operatorDiscoveryResult.getClientSecret();
+    }
+
+    @Override
+    public String getRedirectUri() {
+        return operatorDiscoveryConfig.getOperatorDiscoveryRedirectUri();
+    }
+
+    @Override
+    public String getExpectedIssuer(String actualIssuer) {
```

Comment:
Since the fix in comoyoauth is in code review now, I removed this workaround

(see https://gerrit.c.skunk-works.no/#/c/45471/1)